File: python/upload-fragment-final.py
import http.client

conn = http.client.HTTPSConnection("tenant-my.sharepoint.com")

payload = "{ }"

headers = { 'content-type': "application/json" }

conn.request("PUT", "/alkjl1kjklna", payload, headers)

res = conn.getresponse()
data = res.read()

print(data.decode("utf-8"))

File: ruby/upload-fragment-final.rb
require 'uri'
require 'net/http'

url = URI("https://tenant-my.sharepoint.com/alkjl1kjklna")

http = Net::HTTP.new(url.host, url.port)
http.use_ssl = true
http.verify_mode = OpenSSL::SSL::VERIFY_NONE

request = Net::HTTP::Put.new(url)
request["content-type"] = 'application/json'
request.body = "{ }"

response = http.request(request)
puts response.read_body
